src/ja/user/message/cancel.py

```diff
@@ -59,5 +68,30 @@ def from_dict(cls, property_dict: Dict[str, object]) -> "CancelCommand":
         cls._assert_all_properties_used(property_dict)
         return CancelCommand(label=_label, uid=_uid, config=_config)
 
-    def execute(self, database: ServerDatabase) -> ServerResponse:
-        pass
+    def execute(self, database: ServerDatabase) -> Response:
+        """!
+        Executes this command on the database.
+        @return: A Response informing the user if the job was cancelled or not.
+        """
+        job: Job = None
+        if self.uid is not None:
+            job_entry: DatabaseJobEntry = database.find_job_by_id(self.uid)
+            if job_entry is None:
+                return Response(result_string="Job with uid %s does not exist!" % self.uid,
+                                is_success=False)
+            else:
+                job = job_entry.job
+        if self.label is not None:
+            job = database.find_job_by_label(self.label)
+            if job is None:
+                return Response(result_string="Job with label %s does not exist!" % self.label,
+                                is_success=False)
+        job.status = JobStatus.CANCELLED
```

Comment:
What happens if the job is already finished? Or has been cancelled previously? That should be an error as well.